pkg/templates/notificationtemplates/notification_generator_test.go

```diff
@@ -0,0 +1,103 @@
+package notificationtemplates_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+	"github.com/codeready-toolchain/host-operator/pkg/apis"
+	"github.com/codeready-toolchain/host-operator/pkg/templates/notificationtemplates"
+	testsupport "github.com/codeready-toolchain/toolchain-common/pkg/test"
+
+	"github.com/pkg/errors"
+	uuid "github.com/satori/go.uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+)
+
+func TestCreateOrUpdateResources(t *testing.T) {
+
+	s := scheme.Scheme
+	err := apis.AddToScheme(s)
+	require.NoError(t, err)
+	logf.SetLogger(zap.Logger(true))
+
+	t.Run("ok", func(t *testing.T) {
+		t.Run("create only", func(t *testing.T) {
+			// given
+			namespace := "host-operator" + uuid.NewV4().String()[:7]
+			clt := testsupport.NewFakeClient(t)
+			// verify that notification template resources exist prior to creation
+			for _, templateName := range []string{"userprovisioned"} {
+				template := toolchainv1alpha1.NotificationTemplate{}
+				err = clt.Get(context.TODO(), types.NamespacedName{Namespace: namespace, Name: templateName}, &template)
+				require.Error(t, err)
+				assert.IsType(t, metav1.StatusReasonNotFound, apierrors.ReasonForError(err))
+			}
+			assets := notificationtemplates.NewAssets(notificationtemplates.AssetNames, notificationtemplates.Asset)
+
+			// when
+			err := notificationtemplates.CreateOrUpdateResources(s, clt, namespace, assets)
+
+			// then
+			require.NoError(t, err)
+
+			// verify that notification template resources exist after creation
+			for _, templateName := range []string{"userprovisioned"} {
```

Comment:
instead, you could use `clt.List` to list all existing `NotificationTemplates` and then check the number of returned items and their names and finally verify that they all have non-empty `Subject` and `Content` fields. In that case, when we add more templates, we'll be "forced" to keep this test up-to-date and we will always make sure that all templates are properly set.